github/sub_issue.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2013 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// SubIssueService handles communication with the sub-issue related
+// methods of the GitHub API.
+//
+// Sub-issues help you group and manage your issues with a parent/child relationship.
+//
+// GitHub API docs: https://docs.github.com/rest/issues/sub-issues
+type SubIssueService service
+
+// SubIssue represents a GitHub sub-issue on a repository.
+// Note: As far as the GitHub API is concerned, every pull request is an issue,
+// but not every issue is a pull request. Some endpoints, events, and webhooks
+// may also return pull requests via this struct. If PullRequestLinks is nil,
+// this is an issue, and if PullRequestLinks is not nil, this is a pull request.
+// The IsPullRequest helper method can be used to check that.
+type SubIssue Issue
+
+func (i SubIssue) String() string {
```

Comment:
Please add the appropriate stringify test that you will see throughout this repo.